runtime/lua/vim/lsp/diagnostic.lua

```diff
@@ -22,7 +22,7 @@ end
 ---@param severity lsp.DiagnosticSeverity
 local function severity_lsp_to_vim(severity)
   if type(severity) == 'string' then
-    severity = protocol.DiagnosticSeverity[severity]
+    severity = protocol.DiagnosticSeverity[severity] --- @type integer
```

Comment:
I plan to fix this by fixing the protocol module so maybe best remove this one?

Reply:
Sounds good. I did my best to avoid overlapping work with https://github.com/neovim/neovim/pull/27399 and https://github.com/neovim/neovim/pull/27187, which so far don't seem to conflict with this change.

Although I'm okay with removing the change, what's wrong with a rebase ;) ?

Reply:
Don't worry about #27187, I'll fix whatever I need to in that. Planning on fixing and merging #27399 soon.

Reply:
> Although I'm okay with removing the change, what's wrong with a rebase ;) ?

Ah, sorry I meant this type hint will be redundant after I fix protocol.lua but LuaLS doesn't currently provide hints for unnecessary annotations.

No worries, we can just merge anyway.